src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,92 @@
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo;
+  handleClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, handleClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      setLoading(true);
+      setError(null);
+      try {
+        const fetchedUser = await getUser(todo.userId);
+
+        setUser(fetchedUser);
+      } catch (err) {
+        setError('Не вдалося завантажити користувача');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchUser();
+  }, [todo.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+      <div className="modal-background" onClick={handleClose} />
 
-      {true ? (
-        <Loader />
+      {loading ? (
+        <Loader data-cy="loader" />
+      ) : error ? (
+        <div className="modal-content">
+          <div className="box">
+            <button
+              type="button"
+              className="delete"
+              aria-label="close"
+              data-cy="modal-close"
+              onClick={handleClose}
+            />
+            <p className="has-text-danger">{error}</p>
+          </div>
+        </div>
       ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+        <div className="modal-content">
+          <div className="box">
+            <header className="modal-card-head">
+              <p className="modal-card-title" data-cy="modal-header">
+                Todo #{todo.id}
+              </p>
+              <button
+                type="button"
+                className="delete"
+                aria-label="close"
+                data-cy="modal-close"
+                onClick={handleClose}
+              />
+            </header>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
-          </header>
+            <div className="modal-card-body">
+              <p className="block" data-cy="modal-title">
+                {todo.title}
+              </p>
 
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
+              <p className="block">
+                <strong>Status:</strong>{' '}
+                <span
+                  className={`has-text-${todo.completed ? 'success' : 'danger'}`}
```

Comment:
I see that you are conditionally applying classes based on the `todo.completed` status. This is a perfect use case for the `classnames` library, which can make this code more readable and easier to maintain. You can conditionally apply classes in a more readable way using this library.